src/app/src/components/ClaimedFacilitiesDetails.jsx

```diff
@@ -173,9 +178,39 @@ const InputSection = ({
     );
 
     if (isSelect) {
-        const selectValue = isMultiSelect
-            ? filter(selectOptions, ({ value: option }) => includes(value, option))
-            : find(selectOptions, ['value', value]);
+        const selectValue = (() => {
+            if (!isCreatable && !isMultiSelect) {
+                return find(selectOptions, ['value', value]);
+            }
+
+            if (!isCreatable && isMultiSelect) {
+                return filter(
+                    selectOptions,
+                    ({ value: option }) => includes(value, option),
+                );
+            }
+
+            if (isCreatable && isMultiSelect) {
+                const [
+                    optionsInExistingSet,
+                    newlyCreatedOptionsPrime,
+                ] = partition(
+                    value,
+                    v => includes(map(selectOptions, 'value'), v),
+                );
+
+                const mapStringToOption = s => ({ value: s, label: s });
+
+                return concat(
+                    map(optionsInExistingSet, mapStringToOption),
+                    map(newlyCreatedOptionsPrime, mapStringToOption),
+                );
+            }
+
+            window.console.warn('isCreatable && !isMultiSelect case is not yet implemented');
```

Comment:
We do not currently have a select box with these settings here but I left open the possibility that we could.